pkg/datamover/datamover.go

```diff
@@ -0,0 +1,112 @@
+package datamover
+
+import (
+	"context"
+	"fmt"
+	"github.com/apex/log"
+	snapmoverv1alpha1 "github.com/konveyor/volume-snapshot-mover/api/v1alpha1"
+	"github.com/pkg/errors"
+	velerov1api "github.com/vmware-tanzu/velero/pkg/apis/velero/v1"
+	"golang.org/x/sync/errgroup"
+	"k8s.io/apimachinery/pkg/util/wait"
+	"os"
+	kbclient "sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/client/config"
+	"strconv"
+	"time"
+)
+
+const (
+	// Env vars
+	VolumeSnapshotMoverEnv = "VOLUME_SNAPSHOT_MOVER"
+)
+
+// We expect VolumeSnapshotMoverEnv to be set once when container is started.
+// When true, we will use the csi data-mover code path.
+var dataMoverCase, _ = strconv.ParseBool(os.Getenv(VolumeSnapshotMoverEnv))
+
+// DataMoverCase use getter to avoid changing bool in other packages
+func DataMoverCase() bool {
+	return dataMoverCase
+}
+
+func GetVolumeSnapMoverClient() (kbclient.Client, error) {
+	client, err := kbclient.New(config.GetConfigOrDie(), kbclient.Options{})
+	if err != nil {
+		return nil, err
+	}
+	err = snapmoverv1alpha1.AddToScheme(client.Scheme())
+	if err != nil {
+		return nil, err
+	}
+
+	return client, err
+}
+
+func CheckIfVolumeSnapshotBackupsAreComplete(ctx context.Context, volumesnapshotbackups snapmoverv1alpha1.VolumeSnapshotBackupList) error {
+	eg, _ := errgroup.WithContext(ctx)
+	// update this to configurable timeout
+	timeout := 10 * time.Minute
```

Comment:
Open a bug for this please

Reply:
https://issues.redhat.com/browse/OADP-644